core/src/main/scala/caliban/execution/Executor.scala

```diff
@@ -58,11 +58,11 @@ object Executor {
         case s @ PureStep(value) =>
           value match {
             case EnumValue(v) if mergeFields(currentField, v).collectFirst {
-                  case Field("__typename", _, _, _, _, _, _) => true
+                  case Field("__typename", _, _, _, _, _, _, _) => true
                 }.nonEmpty =>
```

Comment:
I personally don't like many underscores, though I understand that was the case prior to this PR...

How about this notation?
`                  case f: Field if f.name == "__typename" => true`

Reply:
I agree it gets verbose. One concern I have about this change is that it removes exhaustive checks on the case match. I think it’s less of a concern where there are `if` constraints because I think that already disables the check as well.